prime-router/src/main/kotlin/tokens/OktaAuthentication.kt

```diff
@@ -34,97 +32,72 @@ enum class PrincipalLevel {
-         * Otherwise return null.
+         * Perform authentication on a human user.
+         *
+         * @return the claims found in the jwt if the jwt token in [request] is validated.
+         * Return null if not authenticated.
+         * Always performs authentication using Okta, unless running locally.
          */
-        fun getAccessToken(request: HttpRequestMessage<String?>): String? {
-            // RFC6750 defines the access token
-            val authorization = request.headers[HttpHeaders.AUTHORIZATION.lowercase()] ?: return null
-            return authorization.substringAfter("Bearer ", "")
+        fun authenticate(request: HttpRequestMessage<String?>): AuthenticatedClaims? {
+            val accessToken = AuthenticationStrategy.getAccessToken(request)
+            val client = request.headers["client"]
+            return authenticate(accessToken, request.httpMethod, request.uri.path, client)
         }
-    }
 
-    /**
-     * Perform authentication on a human user.
-     *
-     * @return the claims found in the jwt if the jwt token in [request] is validated.
-     * Return null if not authenticated.
-     * Always performs authentication using Okta, unless running locally.
-     */
-    fun authenticate(request: HttpRequestMessage<String?>): AuthenticatedClaims? {
-        val accessToken = getAccessToken(request)
-        return authenticate(accessToken, request.httpMethod, request.uri.path)
-    }
+        /**
+         * Confirm this [accessToken] is a valid Okta token.
+         * [httpMethod] and [path] are just for logging.
+         * Optional [client] is the 'client' header as passed in API POSTs submissions. Only if running local,
+         * use this as the sender in the claims.  Otherwise, ignore [client].   OK if null.
+         */
+        fun authenticate(
+            accessToken: String?,
+            httpMethod: HttpMethod,
+            path: String,
+            client: String? = null,
+        ): AuthenticatedClaims? {
+            if (AuthenticationStrategy.isLocal(accessToken)) {
+                logger.info("Granted test auth request for $httpMethod:$path")
+                val sender = if (client == null)
+                    null
+                else
+                    WorkflowEngine().settings.findSender(client)
+                return AuthenticatedClaims.generateTestClaims(sender)
+            }
 
-    /**
-     * @see [authenticate].
-     */
-    fun authenticate(
-        accessToken: String?,
-        httpMethod: HttpMethod,
-        path: String,
-    ): AuthenticatedClaims? {
-        if (isLocal(accessToken)) {
-            logger.info("Granted test auth request for $httpMethod:$path")
-            return AuthenticatedClaims.generateTestClaims()
-        }
+            // Confirm the token exists.
+            if (accessToken == null) {
+                logger.info("Missing or badly formatted Authorization Header: $httpMethod:$path}")
```

Comment:
Should this be an error or warning?

Reply:
This seems ok.   A missing auth isn't necessarily a sign of a security issue.  Just somebody forgot to include the token.